icons/types/index.d.ts

```diff
@@ -0,0 +1,259 @@
+import * as React from 'react'
+
+export interface IconProps {
+    color?: string
+}
+
+export const IconAdd16: React.FC<IconProps>
```

Comment:
I've plans to generate these using `tsc` and `JsDoc` in the template that generates the icons. But that is quite a lot of changes, so will do that in another PR once this is merged. For now, I've manually listed the icons.